### ISSUE TITLE
EC2 instance cost missing with Elastic Beanstalk when using dynamic "setting" blocks

### ISSUE THREAD

**Issue by dmitriyklyuzov** [infracost/infracost#2548]
When a dynamic "setting" block is used within the `aws_elastic_beanstalk_environment` resource in combination with specifying the EC2 instance type using either the "InstanceType" or the "InstanceTypes" settings, the Instance usage cost is not calculated properly.

### This works
(InstanceTypes without dynamic `settings` blocks)
```
resource "aws_elastic_beanstalk_environment" "this" {
  name        = "test"
  application = "test-app"

  setting {
    namespace = "aws:ec2:instances"
    name      = "InstanceTypes"
    value     = "t3.2xlarge"
  }
}
```
Result: "Instance usage (Linux/UNIX, on-demand, t3.2xlarge)"

### This also works 
(InstanceTypes using a dynamic block)
```
locals {
  settings = [
    {
      namespace = "aws:ec2:instances"
      name      = "InstanceTypes"
      value     = "t3.2xlarge"
    }
  ]
}

resource "aws_elastic_beanstalk_environment" "this" {
  name        = "tf-test-name"
  application = "test-app"

  dynamic "setting" {
    for_each = local.settings
    content {
      namespace = setting.value["namespace"]
      name      = setting.value["name"]
      value     = setting.value["value"]
      resource  = ""
    }
  }
}
```
Result: "Instance usage (Linux/UNIX, on-demand, t3.2xlarge)"

### This does NOT work
(regardless of whether the dynamic block comes before or after the `InstanceTypes` setting)
```
locals {
  vpc_settings = [
    {
      namespace = "aws:ec2:vpc"
      name      = "VPCId"
      value     = "123"
      resource  = ""
    }
  ]
}

resource "aws_elastic_beanstalk_environment" "this" {
  name        = "tf-test-name"
  application = "test-app"

  dynamic "setting" {
    for_each = local.vpc_settings
    content {
      namespace = setting.value["namespace"]
      name      = setting.value["name"]
      value     = setting.value["value"]
      resource  = ""
    }
  }

  setting {
    namespace = "aws:ec2:instances"
    name      = "InstanceTypes"
    value     = "t3.2xlarge"
  }
}
```
Result: "Instance usage (Linux/UNIX, on-demand, )"

### Workaround
Infracost doesn't seem to be able to determine the EC2 instance cost dynamically, but a workaround is to use a JSON Terraform plan as described in [Troubleshooting](https://www.infracost.io/docs/troubleshooting/#terraform-cli):
```
cd path/to/code
terraform init

# Customize this to how you run Terraform
# e.g. if you are using variables you can pass them with -var or -var-file
terraform plan -out tfplan.binary
terraform show -json tfplan.binary > plan.json
infracost breakdown --path plan.json
```

**Comment by hugorut** [comment]
Thanks @dmitriyklyuzov for raising this issue, this does indeed look like a bug. We'll add this to the backlog and work on it when we can get a chance. I'll let you know here when a fix is available. Thanks

**Comment by hugorut** [comment]
Hi @dmitriyklyuzov, I'll see what we can do, but I can't promise that this will be a quick fix as the `dynamic` functionality is quite a complex piece. I'll keep you posted.

**Comment by hugorut** [comment]
Hi @dmitriyklyuzov, I just wanted to let you know that this issue has been fixed in the 0.10.25 release, however, I recommend upgrading to our latest release [0.10.26](https://github.com/infracost/infracost/releases/tag/v0.10.26). 0.10.25 has a critical bug with reporting runs to Infracost Cloud, so I do not advise using that version. If you are using one of our hosted runners (Github/Gitlab app) then you don't need to upgrade you'll automatically have the fix applied and should start seeing accuracy improvements with recent runs.

**Comment by dmitriyzov** [comment]
@hugorut I forgot to reply - thank you for prioritizing and fixing this! (better late than never haha)
